src/java/org/apache/cassandra/db/marshal/DecimalType.java

```diff
@@ -225,7 +225,7 @@ public <V> V fromComparableBytes(ValueAccessor<V> accessor, ByteSource.Peekable
         // but when decoding we don't need that property on the transient mantissa value.
         BigInteger mantissa = BigInteger.ZERO;
         int curr = comparableBytes.next();
-        while (curr != DECIMAL_LAST_BYTE)
+        while (curr != DECIMAL_LAST_BYTE && curr != ByteSource.END_OF_STREAM)
```

Comment:
I wonder if we should only use the `ByteSource.END_OF_STREAM` condition when we expect to need it? I am worried about it possibly changing the behavior in non SAI code paths.

Reply:
I don't think END_OF_STREAM (-1) can be ever used by any byte comparable representation in any other place than end of the stream. @blambov  ?